src/main/java/net/mcreator/ui/blockly/BlocklyEditorType.java

```diff
@@ -19,30 +19,13 @@
 
 package net.mcreator.ui.blockly;
 
-public enum BlocklyEditorType {
+public record BlocklyEditorType(String registryName, String extension, String startBlockName) {
 
-	PROCEDURE("procedures", "ptpl", "event_trigger"), AI_TASK("ai_setup", "aitpl", "aitasks_container"), COMMAND_ARG(
-			"cmd_setup", "cmdtpl", "args_start"), FEATURE("features", "ftpl", "feature_container");
+	public static final BlocklyEditorType PROCEDURE = new BlocklyEditorType("procedures", "ptpl", "event_trigger");
+	public static final BlocklyEditorType AI_TASK = new BlocklyEditorType("aitasks", "aitpl", "aitasks_container");
+	public static final BlocklyEditorType COMMAND_ARG = new BlocklyEditorType("cmdargs", "cmdtpl", "args_start");
+	public static final BlocklyEditorType FEATURE = new BlocklyEditorType("features", "ftpl", "feature_container");
+	public static final BlocklyEditorType JSON_TRIGGER = new BlocklyEditorType("jsontriggers", null, "advancement_trigger");
+	public static final BlocklyEditorType GLOBAL_TRIGGER = new BlocklyEditorType("triggers", null, null);

```

Comment:
Global trigger is not really a blockly editor type so I would still keep string from this one hardcoded

It will be confusing for new code readers to see blockly editor types and see global trigger here where there is no blockly editor for global triggers

Reply:
> and a second one for global triggers (it was required in the end).

I explained why I added it in my long message above. So except if you want an even bigger (and messier) PR to accept both literal Strings and BlocklyEditorType objects, you need to choose between literal Strings for everything or BET objects for everything, including global triggers. However, I can try to change this behaviour in another PR after this one is merged.

Another simple solution might be to simply rename the class to something more global.

Reply:
The only location where it is called is here:

![image](https://user-images.githubusercontent.com/16374228/210326831-5849199f-0777-4ba2-b86b-b409aa25e2d3.png)

I see no reason why here you couldn't use hardcoded literal. Trigger loader does not even have to do anything with this PR as this pr addresses Blockly editors, not a system for loading triggers.

Reply:
The other case is addBlocklyFolder, which is also wrong to be used for triggers, and you have special if there:

![image](https://user-images.githubusercontent.com/16374228/210327030-f564f162-f537-46ad-8736-a19c4f55bd54.png)

So trigger stats could and should be handled by another method and thus eliminating the need for using GLOBAL_TRIGGER  as Blockly editor type

Reply:
Ok fine. However, its registry name will be hardcoded everywhere.